tests/Aspire.Hosting.Kubernetes.Tests/KubernetesPublisherTests.cs

```diff
@@ -9,48 +9,85 @@
-            NullLogger<KubernetesPublisher>.Instance,
-            new DistributedApplicationExecutionContext(DistributedApplicationOperation.Publish));
+    private static readonly Dictionary<string, string> s_expectedFilesCache = new()
+    {
+        ["Chart.yaml"] = ExpectedValues.Chart,
+        ["values.yaml"] = ExpectedValues.Values,
+        ["templates/project1/deployment.yaml"] = ExpectedValues.ProjectOneDeployment,
+        ["templates/project1/configmap.yaml"] = ExpectedValues.ProjectOneConfigMap,
+        ["templates/myapp/deployment.yaml"] = ExpectedValues.MyAppDeployment,
+        ["templates/myapp/service.yaml"] = ExpectedValues.MyAppService,
+        ["templates/myapp/configmap.yaml"] = ExpectedValues.MyAppConfigMap,
+        ["templates/myapp/secret.yaml"] = ExpectedValues.MyAppSecret,
+    };
+
+    public static TheoryData<string> GetExpectedFiles() => new(s_expectedFilesCache.Keys);
+
+    [Theory, MemberData(nameof(GetExpectedFiles))]
+    public async Task PublishAsync_GeneratesValidHelmChart(string expectedFile)
+    {
+        if (!s_publisherHasRun)
+        {
+            // Arrange
+            ArgumentNullException.ThrowIfNull(fixture.TempDirectoryInstance);
+            var options = new OptionsMonitor(
+                new()
+                {
+                    OutputPath = fixture.TempDirectoryInstance.Path,
+                });
+
+            var builder = DistributedApplication.CreateBuilder();
+
+            var param0 = builder.AddParameter("param0");
+            var param1 = builder.AddParameter("param1", secret: true);
+            var param2 = builder.AddParameter("param2", "default", publishValueAsDefault: true);
+            var cs = builder.AddConnectionString("cs", ReferenceExpression.Create($"Url={param0}, Secret={param1}"));
+
+            // Add a container to the application
+            var api = builder.AddContainer("myapp", "mcr.microsoft.com/dotnet/aspnet:8.0")
+                .WithEnvironment("ASPNETCORE_ENVIRONMENT", "Development")
+                .WithHttpEndpoint(env: "PORT")
+                .WithEnvironment("param0", param0)
+                .WithEnvironment("param1", param1)
+                .WithEnvironment("param2", param2)
+                .WithReference(cs)
+                .WithVolume("logs", "/logs")
+                .WithArgs("--cs", cs.Resource);
+
+            builder.AddProject<TestProject>("project1", launchProfileName: null)
+                .WithEnvironment("OTEL_DOTNET_EXPERIMENTAL_OTLP_EMIT_EXCEPTION_LOG_ATTRIBUTES", "true")
+                .WithEnvironment("OTEL_DOTNET_EXPERIMENTAL_OTLP_EMIT_EVENT_LOG_ATTRIBUTES", "true")
+                .WithEnvironment("OTEL_DOTNET_EXPERIMENTAL_OTLP_RETRY", "in_memory")
+                .WithEnvironment("OTEL_DOTNET_EXPERIMENTAL_ASPNETCORE_DISABLE_URL_QUERY_REDACTION", "true")
+                .WithEnvironment("OTEL_DOTNET_EXPERIMENTAL_HTTPCLIENT_DISABLE_URL_QUERY_REDACTION", "true")
```

Comment:
Why do you need this?

Reply:
The test context doesn't appear to set them when you run from cli, but they exist in the output config maps and at run time as they're set by the AppHost aren't they?

Reply:
These are set by the AddProject call. Something funky is going on. See

https://github.com/dotnet/aspire/blob/b3cc78f869a2a709454c78bac7d2ebe228e6f949/tests/Aspire.Hosting.Docker.Tests/DockerComposePublisherTests.cs#L37-L38

https://github.com/dotnet/aspire/blob/b3cc78f869a2a709454c78bac7d2ebe228e6f949/tests/Aspire.Hosting.Docker.Tests/DockerComposePublisherTests.cs#L82-L84

No magic.

Reply:
Test wasn't using the Test App builder, it was using the standard - sorted now :)